functions-python/operations_api/src/feeds_operations/impl/feeds_operations_impl.py

```diff
@@ -0,0 +1,180 @@
+from feeds_operations_gen.models.update_request_gtfs_rt_feed import (
+    UpdateRequestGtfsRtFeed,
+)
+from helpers.database import start_db_session, refresh_materialized_view
+from helpers.query_helper import query_feed_by_stable_id
+from .models.update_request_gtfs_rt_feed_impl import UpdateRequestGtfsRtFeedImpl
+from .request_validator import validate_request
+
+logging.basicConfig(level=logging.INFO)
+
+
+class OperationsApiImpl(BaseOperationsApi):
+    """
+    Implementation of the operations API
+    """
+
+    @staticmethod
+    def detect_changes(
+        feed: Gtfsfeed,
+        update_request_feed: UpdateRequestGtfsFeed | UpdateRequestGtfsRtFeed,
+        impl_class: UpdateRequestGtfsFeedImpl | UpdateRequestGtfsRtFeedImpl,
+    ) -> DeepDiff:
+        """
+        Detect changes between the feed and the update request.
+        """
+        # Normalize the feed and the update request and compare them
+        copy_feed = impl_class.from_orm(feed)
+        # Temporary solution to update the operational status
+        copy_feed.operational_status_action = (
+            update_request_feed.operational_status_action
+        )
+        diff = DeepDiff(
+            copy_feed.model_dump(),
+            update_request_feed.model_dump(),
+            ignore_order=True,
+        )
+        if diff.affected_paths:
+            logging.info(
+                f"Detect update changes: affected paths: {diff.affected_paths}"
+            )
+        else:
+            logging.info("Detect update changes: no changes detected")
+        return diff
+
+    @validate_request(UpdateRequestGtfsFeed, "update_request_gtfs_feed")
+    async def update_gtfs_feed(
+        self,
+        update_request_gtfs_feed: Annotated[
+            UpdateRequestGtfsFeed,
+            Field(description="Payload to update the specified feed."),
+        ],
+    ) -> Response:
+        """Update the specified feed in the Mobility Database.
+        returns:
+            - 200: Feed updated successfully.
+            - 204: No changes detected.
+            - 400: Feed ID not found.
+            - 500: Internal server error.
+        """
+        ...
+        return await self._update_feed(update_request_gtfs_feed, DataType.GTFS)
+
+    @validate_request(UpdateRequestGtfsRtFeed, "update_request_gtfs_rt_feed")
+    async def update_gtfs_rt_feed(
+        self,
+        update_request_gtfs_rt_feed: Annotated[
+            UpdateRequestGtfsRtFeed,
+            Field(description="Payload to update the specified GTFS-RT feed."),
+        ],
+    ) -> Response:
+        """Update the specified GTFS-RT feed in the Mobility Database.
+        returns:
+            - 200: Feed updated successfully.
+            - 204: No changes detected.
+            - 400: Feed ID not found.
+            - 500: Internal server error.
+        """
+        return await self._update_feed(update_request_gtfs_rt_feed, DataType.GTFS_RT)
+
+    async def _update_feed(
```

Comment:
Nit picking: maybe divide _update_feed into smaller chunks for clarity.

async def _update_feed(self, update_request_feed, data_type):
    with session_manager(os.getenv("FEEDS_DATABASE_URL")) as session:
        feed = self._fetch_feed(session, update_request_feed.id, data_type)
        diff = self._detect_and_apply_changes(feed, update_request_feed, session, data_type)

Reply:
Done, Thanks!